File: Stage18/Python/Problem26069.py
import sys
input = sys.stdin.readline

def main():
    N = int(input().strip())
    dancing = set(["ChongChong"])
    
    for _ in range(N):
        a, b = input().split()
        if a in dancing or b in dancing:
            dancing.add(a)
            dancing.add(b)
            
    print(len(dancing))
    
if __name__ == "__main__":
    main()

File: Stage18/Java/Problem26069.java
package Stage18.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.HashSet;
import java.util.Set;
import java.util.StringTokenizer;

public class Problem26069 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        Set<String> dancing = new HashSet<>();
        dancing.add("ChongChong");

        for (int i = 0; i < N; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String a = st.nextToken();
            String b = st.nextToken();

            if (dancing.contains(a) || dancing.contains(b)) {
                dancing.add(a);
                dancing.add(b);
            }
        }

        System.out.println(dancing.size());
        br.close();
    }
}